### PR TITLE
Time Rounds

### PR DESCRIPTION
**Pull request by AntonAsbrink**
Adds time rounds every third round

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 1]
```diff
@@ -358,66 +361,152 @@ void EnemyFactory::Update(double dt)
...
+		}
+	}
+}
+
+void EnemyFactory::timeRound(double dt)
+{
+	if (m_LevelTime > 0)
+	{
+		m_LevelTimer += dt;
+		if (m_LevelTimer >= 1)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: This is a double. Use 1.0

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 2]
```diff
@@ -358,66 +361,152 @@ void EnemyFactory::Update(double dt)
...
+		{
+			m_LevelTimer -= 1;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: This is a double, use 1.0

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 3]
```diff
@@ -358,66 +361,152 @@ void EnemyFactory::Update(double dt)
...
+			m_LevelTime -= 1;
+
+			Entity* enemyGui = m_pScene->GetEntity("enemyGui");
+			if (enemyGui != nullptr)
+			{
+				enemyGui->GetComponent<component::GUI2DComponent>()->GetTextManager()->SetText(std::to_string(m_LevelTime), "enemyGui");
+			}
+
+
+			if (m_LevelTime == 0)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:412]: This is a double. Use EPSILON

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 4]
```diff
@@ -358,66 +361,152 @@ void EnemyFactory::Update(double dt)
...
+		m_EnemiesToSpawn--;
+		m_EnemySlotsLeft--;
 	}
 }
 
 void EnemyFactory::enemyDeath(Death* evnt)
 {
-	if (strcmp(evnt->ent->GetName().substr(0, 5).c_str(), "enemy") == 0)
+	//We don't care about kills on time rounds
+	if (strcmp(evnt->ent->GetName().substr(0, 5).c_str(), "enemy") == 0 && !m_TimeRound)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:531]: Better to use string.find()

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 5]
```diff
@@ -464,10 +553,32 @@ void EnemyFactory::onRoundStart(RoundStart* evnt)
 	m_SpawnTimer = 0.0f;
 	m_EnemiesKilled = 0;
 
+	//First round is level 0 at this point and we want a time round every third level
+	if (m_Level % 3 == 2)
+	{
+		m_TimeRound = true;
+		m_LevelTime = 30 + 5 * m_Level;
+		m_LevelTimer = 0;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: This is a double; use 0.0

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
It was initially unclear what the timer was counting down towards. Some sort of text saying "Survive" (or something) next to the  timer would be helpful

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp (hunk 3 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:412]: no it is not

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp (hunk 4 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:531]: i don't know if it is better and current solution works

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Add the parchment as background on the GUI. And write something like: Survive: X, where X is the timeLeft

**Review by jocke1995 — DISMISSED** [review]

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
Immortal enemy. Please move EnemyFactory::GetInstance().Update(updateRate) up two rows.
Thank you ❤️

**Review by LeoWikstrom — DISMISSED** [review]
👍🏽

**Review by LeoWikstrom — DISMISSED** [review]
Works

File: StortSpelProjekt/Game/src/main.cpp [hunk 6]
```diff
@@ -118,9 +118,9 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR, int nCmdShow)
             {
                 logicTimer -= updateRate;
             }
+            EnemyFactory::GetInstance().Update(updateRate);
             sceneManager->Update(updateRate);
             physics->Update(updateRate);
-            EnemyFactory::GetInstance().Update(updateRate);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/main.cpp:123]: Is this correct? Should enemyFactory update before all the components?

**Review by jocke1995 — DISMISSED** [review]
When the time round begins, I saw the "old" gui first for a second maybe. and then it swapped to the new GUI at the top left.

**Review by LeoWikstrom — APPROVED** [review]
👍🏽

**Review by jocke1995 — APPROVED** [review]

**Pull request merged**